cclient/serialization/tests/test_json.c

```diff
@@ -580,11 +581,40 @@ void test_serialize_store_transactions(void) {
 }
 
 void test_serialize_were_addresses_spent_from(void) {
-  // TODO
+  serializer_t serializer;
+  const char* json_text =
+      "{"
+      "\"command\":\"wereAddressesSpentFrom\","
+      "\"trytes\":[\"" WERE_ADDRESSES_SPENT_FROM_TRYTES "\"]"
+      "}";
+
+  char_buffer_t* serializer_out = char_buffer_new();
+  init_json_serializer(&serializer);
+
+  were_addresses_spent_from_req_t* req = were_addresses_spent_from_req_new();
+  were_addresses_spent_from_req_add(req, WERE_ADDRESSES_SPENT_FROM_TRYTES);
+
+  serializer.vtable.were_addresses_spent_from_serialize_request(
+      &serializer, req, serializer_out);
+
+  TEST_ASSERT_EQUAL_STRING(json_text, serializer_out->data);
+
+  char_buffer_free(serializer_out);
 }
 
 void test_deserialize_were_addresses_spent_from(void) {
-  // TODO
+  serializer_t serializer;
+  init_json_serializer(&serializer);
+  const char* json_text =
+      "{"
+      "\"states\":[\"true\"]"
+      "}";
+
+  were_addresses_spent_from_res_t* res = were_addresses_spent_from_res_new();
+  serializer.vtable.were_addresses_spent_from_deserialize_response(
+      &serializer, json_text, res);
+
+  TEST_ASSERT_EQUAL_STRING("true", were_addresses_spent_from_res_at(res, 0));
```

Comment:
These should be booleans, not strings.
Maybe wait for https://github.com/iotaledger/entangled/pull/221 to be merged so that you can use what he did.